### PR TITLE
Issue 175 chapter 2 master

### PR DESCRIPTION
**Pull request by dsolt**

### CODE REVIEW

File: Chap_Terms.tex [hunk 1]
```diff
@@ -20,23 +20,30 @@ \chapter{PMIx Terms and Conventions}
...
+\item \declareterm{session} 
+refers to a set of resources assigned by the \ac{WLM} that has been 
+reserved for one or more users. 
+A session is identified by a \emph{session ID} that is  
+a unique within the scope of the governing \acp{WLM}.
+Historically, \ac{HPC} sessions have consisted of a static allocation of resources - i.e., a block of resources assigned to a user in response to a specific request and managed as a unified collection. However, this is changing in response to the growing use of dynamic programming models that require on-the-fly allocation and release of system resources. Accordingly, the term \emph{session} in this document refers to a potentially dynamic entity, perhaps comprised of resources accumulated as a result of multiple allocation requests that are managed as a single unit by the \ac{WLM}.
```
Comment (dsolt) [inline Chap_Terms.tex:30]: Notes from WG:  Use of pool was confusing as it made it sound indefinite whether they were really assigned or just available to be assigned.  I think I understand that they are a pool in the sense that they are available to satisfy the needs of job creation, but overall it was deemed clearer to avoid the use of "pool".  

Would prefer to get rid of expressions marked by "-", but did not do that yet in this case.  

"One or more users"... just want to make sure that is really what we want?

File: Chap_Terms.tex [hunk 2]
```diff
@@ -20,23 +20,30 @@ \chapter{PMIx Terms and Conventions}
...
 
-\item \declareterm{namespace} refers to a character string value assigned by the \ac{RM} or launcher (e.g., \code{mpiexec}) to a \textit{job}. All \textit{applications} executed as part of that \textit{job} share the same \emph{namespace}. The \emph{namespace} assigned to each \emph{job} must be unique within the scope of the governing \ac{RM} and often is implemented as a string representation of a numerical job ID. The \emph{namespace} and \emph{job} terms will be used interchangeably throughout the document.
+\item \declareterm{job} refers to a set of one or more \emph{applications} executed as a single invocation by the user within a session with a unique identifier, the \emph{job ID}, assigned by the \ac{RM} or launcher. For example, the command line ``\textit{mpiexec -n 1 app1 : -n 2 app2}'' generates a single \ac{MPMD} job containing two applications. A user may execute multiple \emph{jobs} within a given session, either sequentially or concerrently.
```
Comment (dsolt) [inline Chap_Terms.tex:32]: no change to job

File: Chap_Terms.tex [hunk 3]
```diff
@@ -20,23 +20,30 @@ \chapter{PMIx Terms and Conventions}
...
 
-\item \declareterm{application} refers to a single executable (binary, script, etc.) member of a \emph{job}.
+\item \declareterm{namespace} refers to a character string value assigned by the \ac{RM} to a \textit{job}.  The requester of the \textit{job} (e.g., \code{mpiexec} or a \ac{PMIx} client) may request a specific \emph{namespace} value.  All \textit{applications} executed as part of that \textit{job} share the same \emph{namespace}. The \emph{namespace} assigned to each \emph{job} must be unique within the scope of the governing \ac{RM} and often is implemented as a string representation of the numerical emph{Job ID}. The \emph{namespace} and \emph{job} terms will be used interchangeably throughout the document.
+
```
Comment (dsolt) [inline Chap_Terms.tex:34]: made the "normal" case the first sentence and then clarified that you can also request a specific namespace name for your job, but tried to downplay that since it is not central to what a namespace is or how they normally are identified by most systems.  Also wanted to get rid of a.k.a.'s and .i.e and .e.g. as much as possible.

File: Chap_Terms.tex [hunk 4]
```diff
@@ -20,23 +20,30 @@ \chapter{PMIx Terms and Conventions}
...
+\item \declareterm{application} represents a set of identical, but not necessarily unique,
+execution contexts within a \emph{job}.
```
Comment (dsolt) [inline Chap_Terms.tex:37]: fairly significant change here. Also moved this down in the list.

File: Chap_Terms.tex [hunk 5]
```diff
@@ -20,23 +20,30 @@ \chapter{PMIx Terms and Conventions}
...
 
-\item \declaretermAlt{slot}{slots} refers to an allocated entry for a process. \acp{WLM} frequently allocate entire nodes to a \emph{session}, but can also be configured to define the maximum number of processes that can simultaneously be executed on each node. This often corresponds to the number of hardware \acp{PU} (typically cores, but can also be defined as hardware threads) on the node. However, the correlation between hardware \acp{PU} and slot allocations strictly depends upon system configuration.
+\item \declaretermAlt{slot}{slots} refers to an allocated set of resources assigned to a process.  \acp{WLM} frequently allocate entire nodes to a \emph{session}, but can also be configured to define the maximum number of processes that can simultaneously be executed on each node. This often corresponds to the number of hardware \acp{PU} (typically cores, but can also be defined as hardware threads) on the node. However, the correlation between hardware \acp{PU} and slot allocations strictly depends upon system configuration.
```
Comment (dsolt) [inline Chap_Terms.tex:41]: Really feel allocated set of resources is preferable to "entry".   Otherwise not other changes

File: Chap_Terms.tex [hunk 6]
```diff
@@ -20,23 +20,30 @@ \chapter{PMIx Terms and Conventions}
...
 
 \item \declaretermAlt{peer}{peers} refers to another process within the same \refterm{job}.
 
-\item \declaretermAlt{workflow}{workflows} refers to an orchestrated execution plan frequently involving multiple \emph{jobs} carried out under the control of a \emph{workflow manager} process. An example workflow might first execute a computational job to generate the flow of liquid through a complex cavity, followed by a visualization job that takes the output of the first job as its input to produce an image output.
+\item \declaretermAlt{workflow}{workflows} refers to an orchestrated execution plan frequently involving multiple \emph{jobs} carried out under the control of a \emph{workflow manager}. An example workflow might first execute a computational job to generate the flow of liquid through a complex cavity, followed by a visualization job that takes the output of the first job as its input to produce an image output.
```
Comment (dsolt) [inline Chap_Terms.tex:47]: just got rid of "process" since it doesn't add anything

File: Chap_Terms.tex [hunk 7]
```diff
@@ -120,8 +128,8 @@ \section{Naming Conventions}
 The \ac{PMIx} standard has adopted the following conventions:
 
 \begin{itemize}
-\item \ac{PMIx} constants and attributes are prefixed with \textbf{\code{PMIX}}.
-\item Structures and type definitions are prefixed with \code{pmix}.
+\item \ac{PMIx} constants and attributes are prefixed with \textbf{\code{PMIX_}}.
+\item Structures and type definitions are prefixed with \code{pmix_}.
 \item Underscores are used to separate words in a function or variable name.
```
Comment (dsolt) [inline Chap_Terms.tex:133]: Not sure we discussed this in our latest meetings... do we want this change?

File: Chap_Terms.tex [hunk 8]
```diff
@@ -20,23 +20,30 @@ \chapter{PMIx Terms and Conventions}
...
+\item \declaretermAlt{slot}{slots} refers to an allocated set of resources assigned to a process.  \acp{WLM} frequently allocate entire nodes to a \emph{session}, but can also be configured to define the maximum number of processes that can simultaneously be executed on each node. This often corresponds to the number of hardware \acp{PU} (typically cores, but can also be defined as hardware threads) on the node. However, the correlation between hardware \acp{PU} and slot allocations strictly depends upon system configuration.
 
 \item \declareterm{rank} refers to the numerical location (starting from zero) of a process within the defined scope. Thus, \emph{job rank} is the rank of a process within its \emph{job} and is synonymous with its unqualified \emph{rank}, while \emph{application rank} is the rank of that process within its \emph{application}.
```
Comment (dsolt) [inline Chap_Terms.tex:43]: One major issue I think we still have is that rank is defined in terms of process which is defined as an OS process.  Thoughts on how to fix this?

File: Chap_Terms.tex [hunk 9]
```diff
@@ -20,23 +20,30 @@ \chapter{PMIx Terms and Conventions}
...
 
-\item \declareterm{namespace} refers to a character string value assigned by the \ac{RM} or launcher (e.g., \code{mpiexec}) to a \textit{job}. All \textit{applications} executed as part of that \textit{job} share the same \emph{namespace}. The \emph{namespace} assigned to each \emph{job} must be unique within the scope of the governing \ac{RM} and often is implemented as a string representation of a numerical job ID. The \emph{namespace} and \emph{job} terms will be used interchangeably throughout the document.
+\item \declareterm{job} refers to a set of one or more \emph{applications} executed as a single invocation by the user within a session with a unique identifier, the \emph{job ID}, assigned by the \ac{RM} or launcher. For example, the command line ``\textit{mpiexec -n 1 app1 : -n 2 app2}'' generates a single \ac{MPMD} job containing two applications. A user may execute multiple \emph{jobs} within a given session, either sequentially or concerrently.
```
Comment (kathrynmohror) [inline Chap_Terms.tex:31]: concerrently -> concurrently

File: Chap_Terms.tex [hunk 10]
```diff
@@ -20,23 +20,30 @@ \chapter{PMIx Terms and Conventions}
...
 
-\item \declaretermAlt{slot}{slots} refers to an allocated entry for a process. \acp{WLM} frequently allocate entire nodes to a \emph{session}, but can also be configured to define the maximum number of processes that can simultaneously be executed on each node. This often corresponds to the number of hardware \acp{PU} (typically cores, but can also be defined as hardware threads) on the node. However, the correlation between hardware \acp{PU} and slot allocations strictly depends upon system configuration.
+\item \declaretermAlt{slot}{slots} refers to an allocated set of resources assigned to a process.  \acp{WLM} frequently allocate entire nodes to a \emph{session}, but can also be configured to define the maximum number of processes that can simultaneously be executed on each node. This often corresponds to the number of hardware \acp{PU} (typically cores, but can also be defined as hardware threads) on the node. However, the correlation between hardware \acp{PU} and slot allocations strictly depends upon system configuration.
```
Comment (kathrynmohror) [inline Chap_Terms.tex:40]: "\acp{WLM} frequently allocate entire nodes to a \emph{session}"  I don't know what the macro \acp generates, but this might be grammatically incorrect. Should it be "WLMs frequently allocate" or "A WLM frequently allocates"?

Also, suggest changing "frequently" to  "typically" here, since 'typically' means what is usually done and I think 'frequently'  implies how often this happens (as in a rate)

Comment (dsolt) [reply Chap_Terms.tex:40]: \acp{WLM} will expand to WLMs

I will make the change on frequently vs typically.   Good call.

File: Chap_Terms.tex (hunk 8 above)
Comment (jjhursey) [reply Chap_Terms.tex:43]: Keep it as a process for now, and let the separate discussion on Issue #262 try to address that in a unique PR.

File: Chap_Terms.tex (hunk 7 above)
Comment (jjhursey) [reply Chap_Terms.tex:131]: I think we do want this change.

**Comment by jjhursey** [comment]
With the changes suggested above, I think this looks good. One it has been updated post a PDF here and we can link it from the agenda for the reading.

**Comment by dsolt** [comment]
[pmix-standard-4.0.pdf](https://github.com/dsolt/pmix-standard/files/5245705/pmix-standard-4.0.pdf)

File: Chap_Terms.tex [hunk 11]
```diff
@@ -20,23 +20,30 @@ \chapter{PMIx Terms and Conventions}
...
 
-\item \declareterm{job} refers to a set of one or more \emph{applications} executed as a single invocation by the user within a session with a unique identifier (a.k.a, the \emph{job ID}) assigned by the \ac{RM} or launcher. For example, the command line ``\textit{mpiexec -n 1 app1 : -n 2 app2}'' generates a single \ac{MPMD} job containing two applications. A user may execute multiple \emph{jobs} within a given session, either sequentially or in parallel.
+\item \declareterm{session} 
+refers to a set of resources assigned by the \ac{WLM} that has been 
+reserved for one or more users. 
+A session is identified by a \emph{session ID} that is
```
Comment (dsolt) [inline Chap_Terms.tex:27]: "at least one user". Perhaps this would be better to make it clear that at implementation is not required to support multiple users within a session, for example, if the WLM does not permit it.

File: Chap_Terms.tex (hunk 2 above)
Comment (dsolt) [reply Chap_Terms.tex:32]: actually we changed a.k.a to a clause

File: Chap_Terms.tex [hunk 12]
```diff
@@ -130,9 +136,8 @@ \section{Naming Conventions}
 \item The \code{pmix_} prefix is used to denote function pointer and type definitions.
 \end{itemize}
 
-Users should not use the \textbf{\code{"PMIX"}}, \textbf{\code{"PMIx"}}, or \textbf{\code{"pmix"}} prefixes in their applications or libraries so as to avoid symbol conflicts with current and later versions of the \ac{PMIx} Standard.
+Users shall not use the \textbf{\code{"PMIX"}}, \textbf{\code{"PMIx"}}, or \textbf{\code{"pmix"}} prefixes in their applications or libraries so as to avoid symbol conflicts with \ac{PMIx} implementations.
 
-%%%%%%%%%%%
```
Comment (dsolt) [inline Chap_Terms.tex:135]: we could fix in the other direction... make this about conflicting with the Standard, but refer to the "naming conflicts" rather than symbol conflicts.

File: Chap_Terms.tex [hunk 13]
```diff
@@ -20,27 +20,34 @@ \chapter{PMIx Terms and Conventions}
...
+\item \declareterm{job} refers to a set of one or more \emph{applications} executed as a single invocation by the user within a session with a unique identifier, the \emph{job ID}, assigned by the \ac{RM} or launcher. For example, the command line ``\textit{mpiexec -n 1 app1 : -n 2 app2}'' generates a single \ac{MPMD} job containing two applications. A user may execute multiple \emph{jobs} within a given session, either sequentially or concerrently.
 
-\item \declareterm{application} refers to a single executable (binary, script, etc.) member of a \emph{job}.
+\item \declareterm{namespace} refers to a character string value assigned by the \ac{RM} to a \textit{job}.  The requester of the \textit{job} (e.g., \code{mpiexec} or a \ac{PMIx} client) may request a specific \emph{namespace} value.  All \textit{applications} executed as part of that \textit{job} share the same \emph{namespace}. The \emph{namespace} assigned to each \emph{job} must be unique within the scope of the governing \ac{RM} and often is implemented as a string representation of the numerical emph{Job ID}. The \emph{namespace} and \emph{job} terms will be used interchangeably throughout the document.
```
Comment (jjhursey) [inline Chap_Terms.tex:33]: Per ASC Meeting Oct. 1, 2020:
 * Suggest to remove the sentence "The requester of the \textit{job} (e.g., \code{mpiexec} or a \ac{PMIx} client) may request a specific \emph{namespace} value"
 * Rationale: For tools you can do this via `PMIX_TOOL_NSPACE` (and `PMIX_TOOL_RANK`), but if that namespace is not unique then there are problems. We do not have a mechanism for non-tool PMIx items. Concern about users expecting to be able to set their namespace values and colliding with each other.
 * (If we wanted to put something in) If a request is made for a specific namespace then the caller must handle the possibility that the connection is rejected due to namespace value conflicts. Namespaces are unique in the system.

File: Chap_Terms.tex [hunk 14]
```diff
@@ -20,27 +20,34 @@ \chapter{PMIx Terms and Conventions}
...
 
-\item \declaretermAlt{slot}{slots} refers to an allocated entry for a process. \acp{WLM} frequently allocate entire nodes to a \emph{session}, but can also be configured to define the maximum number of processes that can simultaneously be executed on each node. This often corresponds to the number of hardware \acp{PU} (typically cores, but can also be defined as hardware threads) on the node. However, the correlation between hardware \acp{PU} and slot allocations strictly depends upon system configuration.
+\item \declaretermAlt{slot}{slots} refers to an allocated set of resources assigned to a process.  \acp{WLM} frequently allocate entire nodes to a \emph{session}, but can also be configured to define the maximum number of processes that can simultaneously be executed on each node. This often corresponds to the number of hardware \acp{PU} (typically cores, but can also be defined as hardware threads) on the node. However, the correlation between hardware \acp{PU} and slot allocations strictly depends upon system configuration.
```
Comment (jjhursey) [inline Chap_Terms.tex:44]: Per ASC Meeting Oct. 1, 2020:
 * Colloquially, a `slot` equals one process "slot" on the node
   * `slot` is the number of processes that you are run on a node.
   * It does not mean core or hwthread (or any specific set of resources). A slot is independent of the resource consumption of the process being lanched there.
   * If oversubscription (and it's allowed by the system), then you are assigning more than one process per slot.
 * Maybe rephrase to "A slot refers to used to define the available space for a single process on a node" and "so the number of slots on a node is the number of processes that can run on a node"? (or "within the specified realm" instead of "node")
 * Is it useful to define a term (e.g., "resource set") that is tied to the resources assigned to a process?
 * The working group will need to discuss this topic further before next reading.

File: Chap_Terms.tex [hunk 15]
```diff
@@ -130,8 +136,8 @@ \section{Naming Conventions}
 The \ac{PMIx} standard has adopted the following conventions:
 
 \begin{itemize}
-\item \ac{PMIx} constants and attributes are prefixed with \textbf{\code{PMIX}}.
-\item Structures and type definitions are prefixed with \code{pmix}.
+\item \ac{PMIx} constants and attributes are prefixed with \textbf{\code{PMIX_}}.
+\item Structures and type definitions are prefixed with \code{pmix_}.
```
Comment (jjhursey) [inline Chap_Terms.tex:140]: Per ASC Meeting Oct. 1, 2020:
 * Concern that existing libraries in deployment will have backward compatibility issues. 
   * Because if a user says "PMIXa" - existing libraries only check for "PMIX" thus classifying this attribute as reserved. If we specify this as "PMIX_" then older libraries (per this change) then those deployed libraries would not be able to 
 * We should add a rationale so we don't forget about this at a later date. Something like "all constants/attributes begin with `PMIX_`. However, constants/attributes starting with `PMIX` may be classified as reserved by older libraries. As such we cannot add the `_` in the specification to maintain correctness with implementations of prior standards."

**Comment by jjhursey** [comment]
For the next ASC meeting can you:
 * Close PR #235 and open a new PR based off of this branch?
 * Post a straw poll on that new PR with a PDF when it's finalized.
